home-open-api/src/main/resources/yaml/openapi.yaml

```diff
@@ -1627,3 +2035,244 @@ components:
           example: 12
         address:
           $ref: '#/components/schemas/Address'
+
+    InvitationStatus:
+      description: Indicates the status of an Invitation
+      type: string
+      enum: &invitation_status_values
+        - pending
+        - processing
+        - accepted
+        - declined
+
+    InvitationType:
+      description: Indicates a type of an Invitation
+      type: string
+      enum: &invitation_type_values
+        - cooperation
+        - apartment
+      example: apartment
+
+    ReadInvitation:
+      allOf:
+        - $ref: '#/components/schemas/BaseReadView'
+      type: object
+      discriminator:
+        propertyName: type
+        mapping:
+          base: '#/components/schemas/ReadCooperationInvitation'
+          apartment: '#/components/schemas/ReadApartmentInvitation'
+      properties:
+        type:
+          $ref: '#/components/schemas/InvitationType'
+        email:
+          minLength: 5
+          maxLength: 320
+          pattern: '^([\w!#$%&''*+\/=?{|}~^-]|\.){1,60}@(?!-)(?:[a-zA-Z0-9-]{2,250}+\.)+[a-zA-Z]{2,6}$'
+          type: string
+          example: other1@gmail.com
+        status:
+          $ref: '#/components/schemas/InvitationStatus'
+    ReadCooperationInvitation:
+      allOf:
+        - $ref: '#/components/schemas/ReadInvitation'
+    ReadApartmentInvitation:
+      allOf:
+        - $ref: '#/components/schemas/ReadInvitation'
+      type: object
+      properties:
+        ownership_part:
+          type: number
+          minimum: 0.0001
+          maximum: 1.0
+          multipleOf: 1e-4
+          example: 0.6588
+
+    CreateInvitation:
+      type: object
+      discriminator:
+        propertyName: type
+        mapping:
+          base: '#/components/schemas/CreateCooperationInvitation'
+          apartment: '#/components/schemas/CreateApartmentInvitation'
+      required:
+        - email
+        - type
+      properties:
+        email:
+          minLength: 5
+          maxLength: 320
+          pattern: '^([\w!#$%&''*+\/=?{|}~^-]|\.){1,60}@(?!-)(?:[a-zA-Z0-9-]{2,250}+\.)+[a-zA-Z]{2,6}$'
+          type: string
+          example: other1@gmail.com
+        type:
+          $ref: '#/components/schemas/InvitationType'
+    CreateCooperationInvitation:
+      allOf:
+        - $ref: '#/components/schemas/CreateInvitation'
+    CreateApartmentInvitation:
+      allOf:
+        - $ref: '#/components/schemas/CreateInvitation'
+      type: object
+      required:
+        - ownership_part
+      properties:
+        ownership_part:
+          type: number
+          minimum: 0.0001
+          maximum: 1.0
+          multipleOf: 1e-4
+          example: 0.6588
+
+    UpdateInvitation:
```

Comment:
I think that email also can be updated if the status is pending.

Reply:
> I think that email also can be updated if the status is pending.

Proposed changes have been made